include/envoy/http/async_client.h

```diff
@@ -70,6 +70,16 @@ class AsyncClient {
      * @param reason  failure reason
      */
     virtual void onFailure(const Request& request, FailureReason reason) PURE;
+
+    /**
+     * Called before finalizing upstream span when the request is complete or reset.
+     * @param span a tracing span to fill with extra tags.
+     * @param response_headers the response headers.
+     * @param success whether the request is successful.
+     */
+    virtual void onBeforeFinalizeUpstreamSpan(Envoy::Tracing::Span& span,
+                                              const Http::ResponseHeaderMap* response_headers,
+                                              bool success) PURE;
```

Comment:
What does success mean exactly? Should this actually be `reset`? Could this be removed entirely if we assume that the request was not reset if response_headers is not null?

Reply:
Resolved.